### PR TITLE
Heedls 463 evaluation summary download

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-463

### Description
Added a "Download evaluation summary" button to the bottom of the Reports page that downloads an excel file selecting the currently filtered data.
Also changed the cards to be full width and have the actual values on them as well as the percentages as discussed at show and tell.

### Screenshots
image
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponseBreakdown.cs
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponseBreakdown.cs:20]: Don't think you made this, but is there any particular reason this is nullable? Feels like we should just have this as empty by default.

File: DigitalLearningSolutions.Data.Tests/Services/EvaluationSummaryServiceTests.cs
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Data.Tests/Services/EvaluationSummaryServiceTests.cs:67]: Could we make the `// Given     // When     // Then` all begin with a capital letter?

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Looking good, some very minor tidying to do.

File: DigitalLearningSolutions.Data.Tests/Services/ActivityServiceTests.cs
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data.Tests/Services/ActivityServiceTests.cs:252]: typo in the method name, groud to group

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Centre/Reports/_EvaluationSummaryCard.cshtml
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Views/TrackingSystem/Centre/Reports/_EvaluationSummaryCard.cshtml:23]: I personally think it should be obvious enough with Option - Number - Percentage, but a table may be better? I guess we can always ask at show and tell.

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Centre/Reports/_EvaluationSummaryCard.cshtml
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Views/TrackingSystem/Centre/Reports/_EvaluationSummaryCard.cshtml:15]: Yes, there's still some styling associated with them

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs:213]: No, this is the same date format we use for other downloads, and the best date format for naming files anyway (so that they show in the correct order in file explorer)

File: DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponseBreakdown.cs
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponseBreakdown.cs:20]: Agreed, I see no reason for it being done that way originally. Changed them all to empty lists.

File: DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponses.cs
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponses.cs:9]: Just realised this technically isn't a percentage. Could do with either being renamed or multiplying by 100 to make it a percentage.

**Review by DanBloxham-sw — COMMENTED** [review]
Looks good but I've highlighted one tiny thing I missed before.

File: DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponses.cs
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponses.cs:9]: I'm not really sure what I should name this, it is technically a percentage, just written in decimal form. Googling for a more correct name comes up with that there is no correct name for this. e.g this [thread on Maths StackExchange](https://math.stackexchange.com/questions/1350980/correct-term-for-percentage-in-decimal-form).
Options really are Proportion (not really correct), Percentage (technically correct), DecimalPercentage (not a real term but does describe what we are using, or something more wordy to fully describe it. I've gone with DecimalPercentage for now, but am open to other thoughts.

Multiplying by 100 is incorrect here, it changes the meaning to 100x bigger.

Comment (DanBloxham-sw) [reply DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponses.cs]: I think DecimalPercentage is suitable enough if the value stays as it is. 

This might be a stupid question, but why would multiplying by 100 be incorrect here? It would be converted to an actual percentage, and I don't think its used in any calculations. Is it necessary to keep it as a decimal?

Comment (DanBloxham-sw) [reply DigitalLearningSolutions.Data/Models/TrackingSystem/EvaluationResponses.cs]: Ah right, I think I might have been getting confused with some view models where we do multiply by 100 and do some rounding for display. Fair point. In which case either Percentage or DecimalPercentage works as a name, but I'll approve this and leave it up to you if you have any preference.

**Pull request merged**
